apps/job-service/src/jobs/delayed-jobs.controller.ts

```diff
@@ -11,26 +12,50 @@ import { DelayedJob } from '@terramatch-microservices/database/entities';
 
 @Controller('jobs/v3/delayedJobs')
 export class DelayedJobsController {
+  @Get()
+  @ApiOperation({
+    operationId: 'listDelayedJobs',
+    description: 'Retrieve a list of all delayed jobs.',
+  })
+  @JsonApiResponse({ data: { type: DelayedJobDto } })
+  @ApiException(() => UnauthorizedException, {
+    description: 'Authentication failed.',
+  })
+  async getRunningJobs(
+    @Request() { authenticatedUserId }
+  ): Promise<JsonApiDocument> {
+    const runningJobs = await DelayedJob.findAll({
+      where: {
+        isCleared: 0,
+        createdBy: authenticatedUserId
+      },
+      order: [['createdAt', 'DESC']],
+    });
+
+    const document = buildJsonApi();
+    runningJobs.forEach((job) => {
+      document.addData(job.uuid, new DelayedJobDto(job));
+    });
+    return document.serialize();
+  }
+
   @Get(':uuid')
   @ApiOperation({
     operationId: 'delayedJobsFind',
-    description: 'Get the current status and potentially payload or error from a delayed job.',
+    description:
+      'Get the current status and potentially payload or error from a delayed job.',
   })
   @JsonApiResponse({ data: { type: DelayedJobDto } })
   @ApiException(() => UnauthorizedException, {
     description: 'Authentication failed.',
   })
   @ApiException(() => NotFoundException, {
-    description: 'Job with that UUID not found.'
+    description: 'Job with that UUID not found.',
   })
   async findOne(@Param('uuid') pathUUID: string): Promise<JsonApiDocument> {
-    const job = await DelayedJob.findOne({ where: { uuid: pathUUID }});
+    const job = await DelayedJob.findOne({ where: { uuid: pathUUID } });
     if (job == null) throw new NotFoundException();
 
-    // Note: Since jobs are very generic and we don't track which resources are related to a given
-    // job, there is no effective way to make a policy for jobs until we expand the service to
-    // include an owner ID on the job table.
```

Comment:
Please don't remove this comment without implementing a policy to replace it. Now that we have a `createdBy` on delayed jobs, creating a policy is possible and should potentially be done by this ticket. But if it's not done by this ticket, this comment is still needed as a reminder to do it in the future.

Reply:
Rollback erased.